File: adx-dashboards/scripts/list_dashboards.py
#!/usr/bin/env python3
"""List all ADX dashboards accessible to the current user."""

import asyncio
import json
import sys

import httpx
from azure.identity import DefaultAzureCredential

API_BASE = "https://dashboards.kusto.windows.net"
SCOPE = "https://rtd-metadata.azurewebsites.net/.default"


async def list_dashboards() -> list[dict]:
    credential = DefaultAzureCredential()
    token = credential.get_token(SCOPE)
    
    headers = {
        "Authorization": f"Bearer {token.token}",
        "Accept": "application/json",
    }
    
    async with httpx.AsyncClient() as client:
        response = await client.get(f"{API_BASE}/api/dashboards", headers=headers)
        response.raise_for_status()
        return response.json()


def main():
    try:
        dashboards = asyncio.run(list_dashboards())
        print(json.dumps(dashboards, indent=2))
    except httpx.HTTPStatusError as e:
        print(json.dumps({"error": str(e), "status_code": e.response.status_code}), file=sys.stderr)
        sys.exit(1)
    except Exception as e:
        print(json.dumps({"error": str(e)}), file=sys.stderr)
        sys.exit(1)


if __name__ == "__main__":
    main()

File: adx-dashboard-authoring/scripts/list-dashboards.js
#!/usr/bin/env node
'use strict';

/*
 * List ADX dashboards accessible to the signed-in user via the Dashboard management API.
 * Read-only. Requires an Azure login that DefaultAzureCredential can use (e.g. `az login`).
 */

const { DefaultAzureCredential } = require('@azure/identity');

const API_BASE = 'https://dashboards.kusto.windows.net';
const SCOPE = 'https://rtd-metadata.azurewebsites.net/.default';

async function listDashboards() {
  const credential = new DefaultAzureCredential();
  const token = await credential.getToken(SCOPE);

  const res = await fetch(`${API_BASE}/api/dashboards`, {
    headers: {
      Authorization: `Bearer ${token.token}`,
      Accept: 'application/json',
    },
  });
  if (!res.ok) {
    const body = await res.text().catch(() => '');
    const err = new Error(`HTTP ${res.status} listing dashboards: ${body}`);
    err.status = res.status;
    throw err;
  }
  return res.json();
}

async function main() {
  try {
    const dashboards = await listDashboards();
    process.stdout.write(JSON.stringify(dashboards, null, 2) + '\n');
  } catch (e) {
    process.stderr.write(JSON.stringify({ error: e.message, status: e.status }) + '\n');
    process.exit(1);
  }
}

module.exports = { listDashboards };

if (require.main === module) {
  main();
}
